File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.c


int minimumSum(int num){
    
    
    int arr[4] = {0,0,0,0};
    int minSum = 0;
    int temp = 0;
    
    
    // split numbers
    // Because numbers have four numbers
    for (int i = 0; i < 4; i++){
        
        arr[i] = num % 10;
        num=num/10;
    }
    
    // (Sorting) because we want to sort small to large numbers array's
    
    for (int i = 0; i < 4; i++){
        
        for (int j = i; j < 4; j++){
         
             if (arr[j] < arr[i]){
                 
                temp =  arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }
    return ((arr[0]*10 + arr[2]) + (arr[1]*10 + arr[3]));
}

File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.java
class Solution {
    public int minimumSum(int num) {
        int number1 = num % 10;
        int number2 = (num % 100) / 10;
        int number3 = (num %1000) / 100;
        int number4 = num / 1000;
        int[] arr = new int[]{number1,number2,number3,number4};
        Arrays.sort(arr);
        System.out.print(Arrays.toString(arr));
        int total = arr[3] + arr[0] * 10 + arr[2] + arr[1]*10;
        return total;   
    }
}